ci-operator/step-registry/gcp/provision/bastionhost/gcp-provision-bastionhost-commands.sh

```diff
@@ -0,0 +1,273 @@
+# cannot rely on presence of ${SHARED_DIR}/metadata.json
+echo "${REGION}" >> "${SHARED_DIR}/proxyregion"
+
+gcloud compute instances create "${CLUSTER_NAME}-bastion" \
+  --image=${IMAGE_NAME} \
+  --image-project=${IMAGE_PROJECT} \
+  --metadata-from-file=user-data=/tmp/proxy.ign \
+  --machine-type=${MACHINE_TYPE} \
+  --network=${NETWORK} \
+  --subnet=${CONTROL_PLANE_SUBNET} \
+  --zone=${ZONE_0} \
+  --tags="${CLUSTER_NAME}-bastion"
+
+echo "Created bastion instance"
+echo "Waiting for the proxy service starting running..." && sleep 60s
+
+if [[ -s "${SHARED_DIR}/xpn.json" ]]; then
+  HOST_PROJECT="$(jq -r '.hostProject' "${SHARED_DIR}/xpn.json")"
+  project_option="--project=${HOST_PROJECT}"
+else
+  project_option=""
+fi
+gcloud ${project_option} compute firewall-rules create "${CLUSTER_NAME}-bastion-ingress-allow" \
+  --network ${NETWORK} \
+  --allow tcp:22,tcp:3128,tcp:3129,tcp:5000,tcp:8080 \
+  --target-tags="${CLUSTER_NAME}-bastion"
+cat > "${SHARED_DIR}/bastion-destroy.sh" << EOF
+gcloud compute instances delete -q "${CLUSTER_NAME}-bastion" --zone=${ZONE_0}
+gcloud ${project_option} compute firewall-rules delete -q "${CLUSTER_NAME}-bastion-ingress-allow"
+EOF
+
+if [ X"${DISCONNECTED_NETWORK}" == X"yes" ]; then
+  gcloud ${project_option} compute firewall-rules create "${CLUSTER_NAME}-bastion-egress-allow" --allow='all' --direction=EGRESS --network="${NETWORK}" --target-tags="${CLUSTER_NAME}-bastion"
+  cat >> "${SHARED_DIR}/ssh-bastion-destroy.sh" << EOF
+gcloud ${project_option} compute firewall-rules delete -q "${CLUSTER_NAME}-bastion-egress-allow"
+EOF
+fi
+
+INSTANCE_ID="${CLUSTER_NAME}-bastion"
+echo "Instance ${INSTANCE_ID}"
+
+# to allow log collection during gather:
+# append to proxy instance ID to "${SHARED_DIR}/gcp-instance-ids.txt"
+echo "${INSTANCE_ID}" >> "${SHARED_DIR}/gcp-instance-ids.txt"
+
+gcloud compute instances list --filter="name=('${INSTANCE_ID}')" \
+  --zones "${ZONE_0}" --format json > /tmp/${INSTANCE_ID}-bastion.json
+PRIVATE_PROXY_IP="$(jq -r '.[].networkInterfaces[0].networkIP' /tmp/${INSTANCE_ID}-bastion.json)"
+PUBLIC_PROXY_IP="$(jq -r '.[].networkInterfaces[0].accessConfigs[0].natIP' /tmp/${INSTANCE_ID}-bastion.json)"
+
+# echo proxy IP to ${SHARED_DIR}/proxyip
+echo "${PUBLIC_PROXY_IP}" >> "${SHARED_DIR}/proxyip"
+
+if [ X"${DISCONNECTED_NETWORK}" == X"yes" ]; then
+  PROXY_URL="http://${CLUSTER_NAME}:${PASSWORD}@${PRIVATE_PROXY_IP}:3128/"
+  # due to https://bugzilla.redhat.com/show_bug.cgi?id=1750650 we don't use a tls end point for squid
+
+  cat >> "${CONFIG}" << EOF
+proxy:
+  httpsProxy: ${PROXY_URL}
+  httpProxy: ${PROXY_URL}
+EOF
+fi
+
+if [ X"${PUBLISH_STRATEGY}" == X"Internal" ]; then
+  CLIENT_PROXY_URL="http://${CLUSTER_NAME}:${PASSWORD}@${PUBLIC_PROXY_IP}:3128/"
+  cat > "${SHARED_DIR}/proxy-conf.sh" << EOF
+export http_proxy=${CLIENT_PROXY_URL}
+export https_proxy=${CLIENT_PROXY_URL}
+EOF
+
+  cat >> "${CONFIG}" << EOF
```

Comment:
The same comment in https://github.com/openshift/release/pull/24778 I mentioned, I suggest moving the action that modify install-config.yaml to ipi-conf-gcp setp, so that make the provision bastion step to be more clean.